extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -797,6 +1021,40 @@ protected Properties loadProperties( final RepositoryFile bundle ) {
     }
   }
 
+  /**
+   * Determine if Domain is datasource type metadata.
+   * @param domain
+   * @return
+   */
+  boolean isMetadataDataSource( Domain domain ) {
+    return domain.getLogicalModels() == null || !isDSWDatasource( domain );
+  }
+
+  /**
+   * Determine if Domain is datasource type datasource wizard.
+   * @param domain
+   * @return
+   */
+  boolean isDSWDatasource( Domain domain ) {
+    return domain.getLogicalModels() != null && domain.getLogicalModels().stream().anyMatch( lm ->
+            lm.getProperty( "AGILE_BI_GENERATED_SCHEMA" ) != null
+                    || lm.getProperty( "WIZARD_GENERATED_SCHEMA" ) != null );
+  }
+
+  /**
+   * Adds datasource property to map based on serialized data converted to Domain object.
+   * @param fileMetadata
+   * @param data
+   */
+  protected void addDataSourceType( Map<String, Serializable> fileMetadata, SimpleRepositoryFileData data ) {
```

Comment:
Is the main ask to add to the end of this methods "inFileMetadata"? 

I feel sometimes you can gain insights based on the methods arguments. Otherwise, then all functions will get really long names.

If someone else agrees, I'll change it.

Reply:
I am fine with the way it is

Reply:
> Is the main ask to add to the end of this methods "inFileMetadata"?

The main ask is that a _non-misleading_ name be used. The one I suggested was just an example.

The arguments don't tell us if something else other than the provided argument objects are modified. Are the internal maps modified? Is the repository updated?

I am aware that it is very difficult to find short and clear names.

When I cannot find a shorter, and still clear name, I use a longer one. Clarity is key for me.